src/wallet/wallet.ts

```diff
@@ -2645,9 +2745,24 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
       }
       return arg;
     });
+    // Validate required createTokenOptions fields
+    if (createTokenOptions.amount === undefined || createTokenOptions.amount === null) {
+      throw new Error('createTokenOptions.amount is required and cannot be null or undefined');
+    }
+    if (createTokenOptions.name === undefined || createTokenOptions.name === null) {
+      throw new Error('createTokenOptions.name is required and cannot be null or undefined');
+    }
+    if (createTokenOptions.symbol === undefined || createTokenOptions.symbol === null) {
+      throw new Error('createTokenOptions.symbol is required and cannot be null or undefined');
+    }
+    // If mintAddress is not provided, use the address from the nano contract caller
+    if (!createTokenOptions.mintAddress) {
+      createTokenOptions.mintAddress = address;
+    }
```

Comment:
suggestion(code-style): I suspect this validation could be done internally by zod if the `fullNodeTokenSchema` ( on file `api/schemas/walletApi.ts` ) was implemented using `z.discriminatedUnion` ( see [src/nano_contract/types](https://github.com/HathorNetwork/hathor-wallet-lib/blob/3475450c2444b1c91bc8328bc774e8782bcaee5c/src/nano_contracts/types.ts#L96-L101) ).